File: RentalControl/Components/Models/Address.cs
namespace RentalControl.Components.Models;

public class Address
{
    public Guid Id { get; set; }
    public string Street { get; set; } = string.Empty;
    public string Number { get; set; } = string.Empty;
    public string Neighborhood { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string ZipCode { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public bool Selected { get; set; }

    public List<Tenant> Tenants { get; set; } = [];
    public List<Contract> Contracts { get; set; } = [];
    public List<Guarantor> Guarantors { get; set; } = [];
    
    public string ShortAddress => $"{Street} {Number}, {Neighborhood}";
    public string FullAddress => $"{Street} {Number}, {Neighborhood}, {City}, {State}, {ZipCode}, {Country}";
}

File: api/dto/address.go
package dto

import (
	"github.com/google/uuid"
	"time"
)

type CreateAddressRequest struct {
	Street       string `json:"street" binding:"required"`
	Number       string `json:"number" binding:"required"`
	Neighborhood string `json:"neighborhood" binding:"required"`
	City         string `json:"city" binding:"required"`
	State        string `json:"state" binding:"required"`
	ZipCode      string `json:"zip_code" binding:"required"`
	Country      string `json:"country" binding:"required"`
}

type UpdateAddressRequest struct {
	ID           uuid.UUID `json:"id" binding:"required"`
	Street       string    `json:"street"`
	Number       string    `json:"number"`
	Neighborhood string    `json:"neighborhood"`
	City         string    `json:"city"`
	State        string    `json:"state"`
	ZipCode      string    `json:"zip_code"`
	Country      string    `json:"country"`
}

type AddressResponse struct {
	ID           uuid.UUID `json:"id"`
	Street       string    `json:"street"`
	Number       string    `json:"number"`
	Neighborhood string    `json:"neighborhood"`
	City         string    `json:"city"`
	State        string    `json:"state"`
	ZipCode      string    `json:"zip_code"`
	Country      string    `json:"country"`
	CreatedAt    time.Time `json:"created_at"`
}
